presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -344,6 +357,79 @@ protected RelationType visitShowColumns(ShowColumns showColumns, AnalysisContext
     }
 
     @Override
+    protected RelationType visitShowCreateTable(ShowCreateTable showCreateTable, AnalysisContext context)
+    {
+        QualifiedObjectName tableName = createQualifiedObjectName(session, showCreateTable, showCreateTable.getName());
```

Comment:
Let's re-order the code like this, since there shouldn't be a table handle for views:
```java
QualifiedObjectName tableName = createQualifiedObjectName(session, showCreateTable, showCreateTable.getName());

if (metadata.getView(session, tableName).isPresent()) {
    throw new SemanticException(NOT_SUPPORTED, showCreateTable, "'%s' is a view, not a table", tableName);
}

Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
if (!tableHandle.isPresent()) {
    throw new SemanticException(MISSING_TABLE, showCreateTable, "Table '%s' does not exist", tableName);
}